docs/sources/clients/promtail/logrotation.md

```diff
@@ -0,0 +1,137 @@
+
+In both cases, after log rotation, all new log lines go to the original `error.log` file.
+
+These types are shown in the images below.
+
+### Copy and Truncate
+![block_diagram](./logrotation-copy-and-truncate.png)
+
+### Rename and Create
+![block_diagram](./logrotation-rename-and-create.png)
+
+Both types of log rotation seem to give the same result. However, there are some subtle differences.
+
+(1) favors the `appender` as its descriptor for the original log file `error.log` doesn't change. Therefore, it can keep writing to the same file descriptor. In other words, re-opening the file is not needed.
+
+However (1) has a serious problem if we bring the `tailer` into account. There is a race between truncating the file and the `tailer` finishing reading that log file. Meaning, there is a high chance of the log rotation mechanism truncating the file `error.log` before the `tailer` reads everything from it.
+
+This is where (2) can help. Here, when the log file `error.log` is renamed to `error.log.1`, the `tailer` still holds the file descriptor of `error.log.1`. Therefore, it can continue reading the log file until it's completed. But that comes with the tradeoff: with (2), you have to signal the `appender` to reopen `error.log` (and `appender` should be able to reopen it). Otherwise, it would keep writing to `error.log.1` as the file descriptor won't change. The good news is that most of the popular `appender` solutions (e.g: Syslog, Kubelet, Docker log driver) support reopening log files when they are renamed.
+
+We recommend (2) as that is the one which works well with Promtail (or any similar log scraping agent) without any data loss. Now let's understand how we exactly configure log rotation in different platforms.
+
+## Configure log rotation
+
+Your logs can be rotated by different components depending on where you are running your applications or services. If you are running them on Linux machines, high chance you will be using [`logrotate`](https://man7.org/linux/man-pages/man8/logrotate.8.html) utility. However, if you are running in Kubernetes, it's not that obvious who rotates the logs and interestingly it may depend on what container runtime your Kubernetes cluster is using.
+
+### Non-Kubernetes
+
+As mentioned above, in Linux machines, log rotation is often handled by [`logrotate`](https://man7.org/linux/man-pages/man8/logrotate.8.html) utility.
+
+It supports both types of log rotation explained before.
+
+#### Copy and Truncate
+```
+/var/log/apache2/*.log {
+        weekly
+        maxsize 1G
+        copytruncate
+}
+```
+
+Here `copytruncate` mode works exactly like (1) explained above.
+
+#### Rename and Create **(Recommend)**
+```
+/var/log/apache2/*.log {
+        weekly
+        maxsize 1G
+        create
+}
+```
+Here `create` mode works like (2) explained above. The `create` mode is optional because it's the default mode in `logrotate`.
+
+The configuration for `logrotate` is usually located in `/etc/logrotate/`.
```

Comment:
It looks like lines 74 and 75 are describing `logrotate` in general (and apply to both methods), in which case I'd move these up to line 52, above the headings for Copy and Truncate and Rename and Create.  Having them after the Rename and Create heading implies that these two sentences only apply to the Rename and Create method.